insights/client/archive.py

```diff
@@ -253,11 +242,14 @@ def cleanup_previous_archive(self):
         '''
         Used at the start, this will clean the temporary directory of previous killed runs
         '''
-        archive_glob = os.path.join(constants.insights_tmp_path, 'insights-archive-*')
-        for file in glob.glob(archive_glob):
-            os.path.join('', file)
-            logger.debug("Deleting previous archive %s", file)
-            shutil.rmtree(file, True)
+        try:
+            for directory in os.listdir(constants.insights_tmp_path):
+                directory_path = os.path.join(constants.insights_tmp_path, directory)
+                if directory.startswith(constants.insights_tmp_prefix):
+                    logger.debug("Deleting previous archive %s", directory_path)
+                    shutil.rmtree(directory_path, True)
```

Comment:
As I understand, there may be more than one `insights-client` run at the same time; in that case, because of this code there may be this situation:
- `insights-client` 1 starts: it creates its own `/var/tmp/insights-client-FOO` temporary directory
- `insights-client` 2 starts as well: `cleanup_previous_archive()` cleanups all the existing `/var/tmp/insights-client*` stuff including `/var/tmp/insights-client-FOO`
- `insights-client` 1 dies in sadness

What I'd do is removing only the bits older than 24h: this way even slow `insights-client` run will not suddenly get their own temporary directories removed under their feet, yet still cleaning old leftovers

Reply:
How about adding a restriction that only one instance of "insights-client" is allowed to run at a time?

Reply:
I am not sure if that is a good idea. If we added so, it will limit the following situations:
1. `insights-client` is running the scheduled collection and the administrator wants to check `insights-client` status. 
2. The administrator want to run `collection` and also `malware`.
3. 
I think that a better option is to allow this kind of situation, that was the first idea (before change the things in the temporary files)